src/main/java/org/gridsuite/modification/server/dto/elasticsearch/EquipmentInfos.java

```
@@ -87,6 +87,14 @@ public static Set<VoltageLevel> getVoltageLevels(@NonNull Identifiable<?> identi
         throw NetworkModificationException.createEquipmentTypeUnknown(identifiable.getClass().getSimpleName());
     }
 
+    public static String getEquipmentTypeName(@NonNull Identifiable<?> identifiable) {
+        return identifiable instanceof HvdcLine hvdcLine
+                ? String.format("%s_%s",
+                identifiable.getType().name(),
+                hvdcLine.getConverterStation1().getHvdcType().name())
```

We don't throw exception if getConverterStation1().getHvdcType() != getConverterStation2().getHvdcType() ?
@SlimaneAmar

---

Yes need to throw an exception as said to @ghazwarhili

---

As said to @ghazwarhili this method need to move to powsybl-ws-common !

---

done 

String **getHvdcTypeName**(HvdcLine hvdcLine) will be moved to powsybl-ws-common later, not in this PR.